frontend/javascripts/oxalis/model/sagas/save_saga.js

```diff
@@ -555,65 +579,67 @@ function* applyAndGetRevertingVolumeBatch(
     // Prepare a snapshot of the bucket's current data so that it can be
     // saved in an VolumeUndoState.
     let bucketData = null;
+    const currentPendingOperations = bucket.pendingOperations.slice();
     if (bucket.hasData()) {
       // The bucket's data is currently available.
       bucketData = bucket.getData();
+      // todo: clarify whether getData could already return the merged data
```

Comment:
todo

Reply:
of course, this turned out to be another race condition. when the bucket data already arrived from the back-end, but wasn't yet compressed, this logic was incorrect. I turned the `backendData` property into a promise which is set immediately when the bucket was loaded. If the promise is set, then it's awaited in this saga. I also added a regression test for this which provokes the error case by introducing an artificial delay to the compression.